scenes/backend/backend.gd

```diff
@@ -10,28 +10,89 @@ var test_password = "meowwwwww"
 
 var is_connected = false
 
+# nakama
 var scheme = "https"
 var host = "nakama.projectlam.org"
 var port = 7352
 var server_key = "projectlam"
 
+# song loading
+var remote_songs_json_url := ""
+var remote_songs_info: Dictionary = {}
+signal remote_songs_json_received
+var skip_remote_json_load := false
+var remote_songs_json_initialized := false
+const RTFILE_RQ_SCENE := preload("res://scenes/networking/remote_file_access/remote_file_request.tscn")
+var song_json_file_request
+var song_remote_access: RemoteFileAccess
+
+
 @onready var nConnectionFailedDialog: Control = find_child("ConnectionFailedDialog")
 @onready var nConnectionFailedDialog_AnimationPlayer: AnimationPlayer = nConnectionFailedDialog.get_node("AnimationPlayer")
 
 @onready var nLoginFailedDialog: Control = find_child("LoginFailedDialog")
 @onready var nLoginFailedDialog_AnimationPlayer: AnimationPlayer = nLoginFailedDialog.get_node("AnimationPlayer")
 
+@onready var ui_node := $CanvasLayer
+
 signal full_initialization
 signal received_match_presence(p_presence: NakamaRTAPI.MatchPresenceEvent)
 signal received_match_state(p_state)
 signal peer_connected(id : int)
 
+
+func _ready():
+	init_remote_songs_json_url()
+	
+	if skip_remote_json_load:
+		# skip
+		remote_songs_json_initialized = true
+	call_deferred("init_json_request")
+	
+	await multiplayer_init_async()
+	if not remote_songs_json_initialized:
+		await remote_songs_json_received
+	
+	
+	_is_fully_initialized = true
+	print("Backend fully initialized.")
+	emit_signal("full_initialization")
+
+
+func _process(delta: float) -> void:
+	nLoginFailedDialog.pivot_offset.x = nLoginFailedDialog.get_rect().size.x/2
+	nLoginFailedDialog.pivot_offset.y = nLoginFailedDialog.get_rect().size.y/2
+	
+	nLoginFailedDialog.pivot_offset.x = nLoginFailedDialog.get_rect().size.x/2
+	nLoginFailedDialog.pivot_offset.y = nLoginFailedDialog.get_rect().size.y/2
+
+
+func init_json_request():
+	if skip_remote_json_load:
+		# skip
+		remote_songs_json_initialized = true
+		return
+	
+	song_json_file_request = RTFILE_RQ_SCENE.instantiate()
+	song_json_file_request.target_url = remote_songs_json_url
+	song_json_file_request.request_completed_string.connect(_on_song_json_file_received)
+	song_remote_access = RemoteFileAccess.new()
+	song_remote_access.parent_url = remote_songs_json_url.get_base_dir()
+	add_child(song_json_file_request)
+
+
+func init_remote_songs_json_url():
+	# TODO : add java calls here.
```

Comment:
The host html page should be able to set the songs URL, here's how I've got it hooked up right now:
```gdscript
func init_remote_songs_json_url():
	var os_name = OS.get_name()
	if os_name == 'HTML5' || os_name == 'Web':
		remote_songs_json_url = JavaScriptBridge.eval("godotGame.getSongsManifest();", true)
	else:
		remote_songs_json_url = "http://127.0.0.1/songs.json"
	print("Remote songs manifest: ", remote_songs_json_url)
```

And in the host page (for example):
```js
globalThis.godotGame = {
	getSongsManifest: function (): string {
		return 'http://127.0.0.1:5173/songs.json';
	}
};
```

Reply:
Yea that's exactly how it's meant to be used.